modules/ingester/instance.go

```diff
@@ -149,12 +166,66 @@ func newInstance(instanceID string, limiter *Limiter, writer tempodb.Writer, l *
 }
 
 func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) error {
+	var errors util.MultiError
+
+	var maxLiveTracesErrorCount int
+	var traceTooLargeErrorCount int
+	var notValidTraceIDErrorCount int
+	var internalErorCount int
+
 	for j := range req.Traces {
-		err := i.PushBytes(ctx, req.Ids[j].Slice, req.Traces[j].Slice)
+		traceID := req.Ids[j].Slice
+		spanCount := 0
+		var spanCountErr error
+		if len(req.SpanCounts) > 0 {
+			spanCountByte := util.TraceIDToHexString(req.SpanCounts[j].Slice)
+			spanCount, spanCountErr = strconv.Atoi(spanCountByte)
+			if spanCountErr != nil {
+				spanCount = 0
+			}
+		}
+
+		err := i.PushBytes(ctx, traceID, req.Traces[j].Slice)
+
 		if err != nil {
-			return err
+			errorMessage := err.Error()
+			if strings.Contains(errorMessage, overrides.ErrorPrefixLiveTracesExceeded) {
+				maxLiveTracesErrorCount += spanCount
+				// only log one of each occurrence
+				if !strings.Contains(errors.Error(), overrides.ErrorPrefixLiveTracesExceeded) {
+					errors.Add(err)
+				}
+			} else if strings.Contains(errorMessage, overrides.ErrorPrefixTraceTooLarge) {
+				traceTooLargeErrorCount += spanCount
+				if !strings.Contains(errors.Error(), overrides.ErrorPrefixTraceTooLarge) {
+					errors.Add(err)
+				}
+			} else if strings.Contains(errorMessage, ("not a valid traceid")) {
+				notValidTraceIDErrorCount += spanCount
+				if !strings.Contains(errors.Error(), "not a valid traceid") {
+					errors.Add(err)
+				}
+			} else {
+				internalErorCount += spanCount
+				if !strings.Contains(errors.Error(), err.Error()) {
+					errors.Add(err)
+				}
+			}
 		}
 	}
+
+	if len(errors) > 0 {
+		discardCountMessage := fmt.Sprintf(
```

Comment:
if we're going to pass back rich information like this we should just change the `PushResponse` method to have these fields.

Reply:
Yeah I originally had tried doing this but the response object never gets back to the distributor and I just assumed that was not possible. I will try to figure out what I did wrong there. Is there any additional things I need to do besides changing the protos for the response?